### DISCUSSION TITLE
Restarting when installing to a new version

### DISCUSSION THREAD

**Discussion by ghost** [WinMerge/winmerge#1854] [General]
Why during a silent install using /NORESTART from older version to a newer version does Windows still restart?

**Comment by sdottaka** [comment]
I couldn't reproduce that restarting. When you say "restart", do you mean restarting the OS or restarting Explorer?

**Comment by ghost** [comment]


**Reply by sdottaka** [reply]
Thank you for answering.
I have tried the following but have not been able to reproduce the OS reboot.

1. Install version 2.16.28 as below.
~~~
WinMerge-2.16.28-x64-Setup.exe /norestart /silent
~~~
2. Right-click the file in Explorer to display the context menu. (Explorer loads and locks ShellExtensionX64.dll.)

3. Install version 2.16.30 as below. (At the end of the installation, you will be asked to restart Explorer, but the OS will not be restarted.)
~~~
WinMerge-2.16.30-x64-Setup.exe /norestart /silent
~~~

**Comment by ghost** [comment]


**Reply by sdottaka** [reply]
Any version requires a restart of Explorer if Explorer already loaded the ShellExtension DLL during installation.
Versions 2.16.28 and earlier would sometimes fail to install ShellExtension when multiple Explorer processes existed, and version 2.16.30 attempts to fix that issue.
